### PR TITLE
Tree Manager tests

### PR DESCRIPTION
**Pull request by Unraveler**
- Add more detailed tests to TreeManager component
- Removes Enzyme
- Still has some warnings running the tests, to check later for a solution
- Skipped one test due to not understanding why the test is not working. To come back to it later
- skipped two older tests that were throwing console warnings. Those will be dealt with in their own branch

### CODE REVIEW

File: frontend/src/Components/MapContextForm/AttributesForm/AttributesForm.spec.tsx [hunk 1]
```diff
@@ -1,21 +1,22 @@
...
-import { shallow } from 'enzyme';
+import { render, waitFor } from '@testing-library/react';
 import React from 'react';
 import { AttributesForm } from './AttributesForm';
 
 
 describe('AttributesForm component', () => {
   const requiredProps = {};
 
-  const getComponent = (props?:any) => shallow((
```
Comment (MrSnyder) [inline frontend/src/Components/MapContextForm/AttributesForm/AttributesForm.spec.tsx:10]: as this renders the component, should it be called `renderComponent`?

File: frontend/src/Components/MapContextForm/LayerAttributesForm/LayerAttributesForm.spec.tsx [hunk 2]
```diff
@@ -1,21 +1,23 @@
 import '@testing-library/jest-dom';
-import { shallow } from 'enzyme';
+import { render, waitFor } from '@testing-library/react';
 import React from 'react';
 import { LayerAttributesForm } from './LayerAttributesForm';
 
 
 describe('LayerAttributesForm component', () => {
   const requiredProps = {};
```
Comment (MrSnyder) [inline frontend/src/Components/MapContextForm/LayerAttributesForm/LayerAttributesForm.spec.tsx:9]: as this renders the component, should it be called renderComponent?

File: frontend/src/Components/Shared/TreeManager/TreeManager.spec.tsx [hunk 3]
```diff
@@ -1,21 +1,461 @@
...
+      expanded: true,
+      isLeaf: true,
+    },
+  ];
 
-  const getComponent = (props?:any) => shallow((
+  const requiredProps = {
+    treeData: mockedTreeData,
+  };
+
```
Comment (MrSnyder) [inline frontend/src/Components/Shared/TreeManager/TreeManager.spec.tsx:63]: as this renders the component, should it be called renderComponent?

File: frontend/src/Components/Shared/TreeManager/TreeManager.spec.tsx [hunk 4]
```diff
@@ -1,21 +1,461 @@
...
+  });
+
+  // NOTE: This component loads the tree data as soon as the tree data is ready.
+  // This fetch will trigger a state event (in this case setTreeData()).
+  // Because of this, we need to simulate this behaviour in an asynchronous fashion, hence the waitFor is used for all
+  // the following tests.
+  // After an event that triggers a state chnage, this is also being made.
+  // Without this, the tests will pass but a couple of warnings about state changes should be
+  // encapsulated in an act() method will appear
+
```
Comment (MrSnyder) [inline frontend/src/Components/Shared/TreeManager/TreeManager.spec.tsx:84]: this tests the rendering  and then if the returned wrapper is defined
can we stay with the common (terrestris) practice, of testing if the component is defined in a separate test, then rendering?
https://github.com/terrestris/react-geo/blob/master/src/LayerTree/LayerTree.spec.tsx

File: frontend/src/Components/Shared/TreeManager/TreeManager.spec.tsx [hunk 5]
```diff
@@ -1,21 +1,461 @@
...
+  it('does not render tree nodes when tree data does not contains information', async() => {
+    const { container } = getComponent({ treeData: [] });
+    await waitFor(() =>{
+      const treeFormFieldTree = container.querySelectorAll('.tree-manager-tree');
+      const treeFormFieldNode = treeFormFieldTree.item(0)?.querySelectorAll('.tree-manager-node');  
+      expect(treeFormFieldNode?.length).toBe(0);
+    });
+  });
+
```
Comment (MrSnyder) [inline frontend/src/Components/Shared/TreeManager/TreeManager.spec.tsx:112]: suggestion: 'renders tree nodes when data contains information' (the tree is rendered in any case, as the first test showed)

File: frontend/src/Components/Shared/TreeManager/TreeManager.spec.tsx [hunk 6]
```diff
@@ -1,21 +1,461 @@
...
+    const settingsNode2 = nodeSettings.item(1)?.querySelector('button');
+    expect(screen.queryByText('Add new layer group')).toBeNull(); 
+    fireEvent.click(settingsNode1 as Element);
+    fireEvent.click(settingsNode2 as Element);
+    await waitFor(() => {
+      expect(screen.queryByText('Add new layer group')).toBeDefined();
+    });
+  });
+
+  it('renders a modal when clicking to add a new node on the options menu and attributeContainer ' +
```
Comment (MrSnyder) [inline frontend/src/Components/Shared/TreeManager/TreeManager.spec.tsx:249]: we don't need the modal in our application, so shouldn't we remove it?
YAGNI: https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it

File: frontend/src/Components/Shared/TreeManager/TreeManager.spec.tsx [hunk 7]
```diff
@@ -1,21 +1,461 @@
...
+    await waitFor(async() => {
+      const contextMenuFirstAction = screen.getByText('Add new layer group');
+      fireEvent.click(contextMenuFirstAction);
+      await waitFor(() => {
+        expect(screen.getByText('Add Node').parentElement?.className).toBe('ant-modal-header');
+      });
+    });
+      
+  });
+
```
Comment (MrSnyder) [inline frontend/src/Components/Shared/TreeManager/TreeManager.spec.tsx:265]: YAGNI

**Review by MrSnyder — APPROVED** [review]
Looks good, feel free to merge. I provided some comments.

File: frontend/src/Components/Shared/TreeManager/TreeManager.spec.tsx (hunk 6 above)
Comment (Unraveler) [reply frontend/src/Components/Shared/TreeManager/TreeManager.spec.tsx:249]: I have remove the modal from the Tree Manager component as well

**Review by MrSnyder — APPROVED** [review]
Nice. Feel free to merge.

**Pull request merged**